File: react-native/client/android/src/main/java/expo/modules/coinbasewalletsdkexpo/records/ActionRecord.kt
package expo.modules.coinbasewalletsdkexpo.records

import com.coinbase.android.nativesdk.message.request.Action
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class ActionRecord : Record {
    @Field
    var method: String = ""

    @Field
    var paramsJson: String = "{}"

    @Field
    var optional: Boolean = false
}

val ActionRecord.asAction: Action
    get() {
        return Action(
            method = this.method,
            paramsJson = this.paramsJson,
            optional = this.optional
        )
    }

File: react-native/client/ios/Records/ActionRecord.swift
import ExpoModulesCore
import CoinbaseWalletSDK
import Foundation

struct ActionRecord : Record {
    @Field
    var method: String

    @Field
    var paramsJson: String

    @Field
    var optional: Bool
}

extension ActionRecord {
    var asAction: Action {
        let paramsJson = self.paramsJson.data(using: .utf8)!
        let params = try! JSONSerialization.jsonObject(with: paramsJson) as! [String: Any]
        return Action(method: self.method, params: params)
    }
}
